Lombiq.Hosting.Tenants.Management/Controllers/ShellSettingsEditorController.cs

```diff
@@ -45,6 +49,12 @@ public async Task<IActionResult> Edit(ShellSettingsEditorViewModel model)
 
         var settingsDictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(model.Json);
         var newSettings = new Dictionary<string, string>();
+
+        foreach (var key in settingsDictionary.Keys.Where(key => string.IsNullOrEmpty(settingsDictionary[key])))
+        {
+            settingsDictionary[key] = null;
+        }
```

Comment:
Is the intent here to convert empty string into `null`s? Why is that necessary?

Also, this can be a foreach on the dictionary, no need to foreach on the keys, then fetch the value for the key.